File: frontend/src/types/job.ts
export type JobEditorMode = "verify" | "edit";

export type EmploymentType =
    | "full_time"
    | "part_time"
    | "contract"
    | "internship"
    | "temporary"
    | "other";

export interface Job {
    id: number;

    company: string;
    title: string;
    location: string;

    employment_type: EmploymentType | null;

    experience_min: number | null;
    experience_max: number | null;

    salary_min: number | null;
    salary_max: number | null;
    currency: string | null;

    required_skills: string[];
    preferred_skills: string[];

    education: string | null;
    application_url: string | null;

    raw_description: string;

    verification_status: "PENDING" | "VERIFIED";

    application_status:
    | "SAVED"
    | "APPLIED"
    | "INTERVIEW"
    | "OFFER"
    | "REJECTED"
    | "WITHDRAWN";

    applied_date: string | null;

    created_at: string;
    updated_at: string;
}

export interface JobFormData {
    company: string;
    title: string;
    location: string;

    employment_type: EmploymentType | "";

    experience_min: string;
    experience_max: string;

    salary_min: string;
    salary_max: string;
    currency: string;

    required_skills: string[];
    preferred_skills: string[];

    education: string;
    application_url: string;

    raw_description: string;
}

File: backend/app/models/job.py
import enum
from sqlalchemy import String, Text, DateTime, func, Enum as SQLEnum, ForeignKey, ARRAY
from sqlalchemy.orm import Mapped, mapped_column
from datetime import datetime
from app.core.database import Base

class ApplicationStatus(str, enum.Enum):
    SAVED = "Saved"
    APPLIED = "Applied"
    INTERVIEW = "Interview"
    OFFER = "Offer"
    REJECTED = "Rejected"
    WITHDRAWN = "Withdrawn"

class VerificationStatus(str, enum.Enum):
    PENDING = "Pending"
    VERIFIED = "Verified"

class Job(Base):
    __tablename__ = "jobs"

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), index=True, nullable=False)
    title: Mapped[str | None] = mapped_column(String(255), nullable=True)
    company: Mapped[str | None] = mapped_column(String(255), nullable=True)
    location: Mapped[str | None] = mapped_column(String(255), nullable=True)
    job_url: Mapped[str | None] = mapped_column(String(1000), nullable=True)
    raw_description: Mapped[str] = mapped_column(Text, nullable=False)
    required_skills: Mapped[list[str]] = mapped_column(ARRAY(String), default=list, nullable=False)
    
    application_status: Mapped[ApplicationStatus] = mapped_column(
        SQLEnum(ApplicationStatus),
        default=ApplicationStatus.SAVED,
        nullable=False
    )
    verification_status: Mapped[VerificationStatus] = mapped_column(
        SQLEnum(VerificationStatus),
        default=VerificationStatus.PENDING,
        nullable=False
    )
    
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)